src/main/java/spoon/support/compiler/SpoonProgress.java

```diff
@@ -0,0 +1,38 @@
+package spoon.support.compiler;
+
+/**
+ * is the interface to follow the progress of the creation of the Spoon model.
+ */
+public interface SpoonProgress {
+
+	enum Process {
+		COMPILE,
+		COMMENT,
+		MODEL,
+		IMPORT,
+		PROCESS,
+		PRINT
+	}
+
+	/**
+	 * is called when a new process is started
+	 * @param process the started process
+	 */
+	void start(Process process);
+
+	/**
+	 * is called when a step in the precess is started
+	 * @param process the current process
+	 * @param task the task that has been processed
+	 * @param taskId the task id
+	 * @param nbTask the number of task in the process
+	 */
+	void step(Process process, String task, int taskId, int nbTask);
```

Comment:
Am I wrong to say that everywhere in your usage, the task is in fact the filename of the file being processed?

Reply:
for the moment it contains only the absolute path of the file.

Reply:
So don't you think that the parameter should be renamed then?

Reply:
I mean: it's not the task, it's at best a description. The task itself is given by the process parameter.

Reply:
it is the task of the process.